src/components/container/SwellingCalendar/SwellingCalendar.tsx

```diff
@@ -0,0 +1,177 @@
+import { useEffect, useState } from "react";
+import MyDataHelps, { Guid, SurveyAnswer, SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+import React from "react";
+import surveyResults from "../../../helpers/get-survey-answers";
+import Calendar from "../../presentational/Calendar/Calendar";
+import { parseISO, startOfDay, startOfMonth } from "date-fns";
+import { Button, CalendarDay, CalendarDayStateConfiguration, DateRangeNavigator, LoadingIndicator } from "../../presentational";
+import { previewSwellingData } from "./SwellingCalendar.previewdata";
+import language from "../../../helpers/language";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+export type SwellingCalendarPreviewState = "WithData" | "NoData" | "Loading" | "Live";
+
+export interface SwellingCalendarProps {
+    surveyName : string,
+    dateRecordedResultIdentifier : string, 
+    severityResultIdentifier : string, 
+    previewState? : SwellingCalendarPreviewState,
+}
+
+interface SwellingDay {
+    dateRecorded : string,
+    severity : string
+}
+
+export default function (props: SwellingCalendarProps) {
+    const surveyAnswerQuery : SurveyAnswersQuery = {
+        surveyName : props.surveyName,
+        resultIdentifier : [props.dateRecordedResultIdentifier, props.severityResultIdentifier]
+    }
+    const [datePagerMonth, setMonth] = useState<Date>(startOfMonth(new Date()));
+    const [data, setData] = useState<Map<string, SwellingDay> | undefined>();
+
+    async function initialize(){
+        if (!data){
+            if (props.previewState && props.previewState == "NoData")
```

Comment:
`props.previewState == "NoData"` should be enough here.  You don't need the initial `props.previewState &&` part.  Also, while I believe Typescript handles it okay, I still prefer to use `===` when performing exact comparisons to avoid any ambiguity.